artifactory/utils/golang/project/project.go

```diff
@@ -239,15 +239,18 @@ func (project *goProject) getId() string {
 // Read go.mod file and add it as an artifact to the build info.
 func (project *goProject) readModFile() error {
 	var err error
-	project.projectPath, err = cmd.GetProjectRoot()
-	if err != nil {
-		return errorutils.CheckError(err)
+	if project.projectPath == "" {
+		project.projectPath, err = cmd.GetProjectRoot()
+		if err != nil {
+			return errorutils.CheckError(err)
+		}
 	}
 
 	modFilePath := filepath.Join(project.projectPath, "go.mod")
 	modFile, err := os.Open(modFilePath)
 	if err != nil {
-		return errorutils.CheckError(err)
+		log.Info("Could not find go.mod in ", project.projectPath)
```

Comment:
There could be a potential situation where the file does exists, but it cannot be opened due to permissions issue.
In case of a permission issue, an error should be thrown. Let's have the code differentiate between the two scenarios.